packages/components/src/unit-control/utils.ts

```diff
@@ -302,17 +309,17 @@ export const useCustomUnits = ( { units, availableUnits, defaultValues } ) => {
  * accurately displayed in the UI, even if the intention is to hide
  * the availability of that unit.
  *
- * @param {number|string}            currentValue Selected value to parse.
- * @param {string}                   legacyUnit   Legacy unit value, if currentValue needs it appended.
- * @param {Array<WPUnitControlUnit>} units        List of available units.
+ * @param  currentValue Selected value to parse.
+ * @param  legacyUnit   Legacy unit value, if currentValue needs it appended.
+ * @param  units        List of available units.
  *
- * @return {Array<WPUnitControlUnit>} A collection of units containing the unit for the current value.
+ * @return A collection of units containing the unit for the current value.
  */
 export function getUnitsWithCurrentUnit(
-	currentValue,
-	legacyUnit,
-	units = ALL_CSS_UNITS
-) {
+	currentValue: number | string,
+	legacyUnit: string | undefined,
+	units: Array< WPUnitControlUnit > | false = ALL_CSS_UNITS
+): Array< WPUnitControlUnit > | false {
```

Comment:
Using `false` as a valid value for `units` and also as a potential return value of the function feels a bit weird (I know the implementation was there before this PR, and in fact, it's thanks to using TypeScript that we can spot situations like this one).

I can see that the same pattern happens in the `hasUnits` function — this is probably something that we should tackle separately in a follow-up PR. WDYT?

Reply:
I thought the same, it does feel a bit weird in these functions. I _think_ we'll still need to support `theme.json` providing `false` as a valid value to switch off units support, but there's likely a neater way of handling it than having these functions use it as an expected type.

> this is probably something that we should tackle separately in a follow-up PR.

Agreed, it'll probably be easier to do as a follow-up as the implementation might need to change slightly.